forwarder/kernel-forwarder/pkg/kernelforwarder/local.go

```diff
@@ -56,26 +59,44 @@ func (k *KernelForwarder) createLocalConnection(crossConnect *crossconnect.Cross
 	runtime.LockOSThread()
 	defer runtime.UnlockOSThread()
 
-	srcName := crossConnect.GetSource().GetMechanism().GetParameters()[common.InterfaceNameKey]
-	dstName := crossConnect.GetDestination().GetMechanism().GetParameters()[common.InterfaceNameKey]
+	rand.Seed(time.Now().UnixNano())
+
+	var err error
+	var tempName string
 	var srcNetNsInode string
 	var dstNetNsInode string
-	var err error
+	srcName := crossConnect.GetSource().GetMechanism().GetParameters()[common.InterfaceNameKey]
+	dstName := crossConnect.GetDestination().GetMechanism().GetParameters()[common.InterfaceNameKey]
+
+	/* In case source and destination names are the same, use temporary values during the setup of the VETH interfaces */
+	if dstName == srcName {
+		tempName = srcName
+		srcName = fmt.Sprintf("nsm%d", rand.Uint32())
```

Comment:
Also, should we use here something like this?
```suggestion
		srcName = fmt.Sprintf("%s%d", srcName, rand.Uint32())
```

Reply:
@denis-tingajkin The reason I chose rand() instead of something else is that rand() makes it almost impossible for any conflicts to occur.

Nevertheless, those temporary names are used only during the actual creation of the VETH pair. Once an interface is moved to its destination namespace, it's renamed and we are back to normal.

Reply:
Adding `srcName` will make it even more impossible IMO.